geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/AcceptorImpl.java

```diff
@@ -1455,22 +1478,24 @@ private void handleNewClientConnection(final Socket socket) throws IOException {
             "Rejected connection from {} because current connection count of {} is greater than or equal to the configured max of {}",
             new Object[] {socket.getInetAddress(), curCnt,
                 maxConnections});
+
         try {
           refuseHandshake(socket.getOutputStream(),
               String.format("exceeded max-connections %s",
                   maxConnections),
-              REPLY_REFUSED);
+              REPLY_REFUSED, ioFilter, socket);
```

Comment:
since you are now passing "socket" into refuseHandshake, you no longer need to also pass in socket.getOutputStream()

Reply:
updated